### PR TITLE
KTOR-7415 Document how to use embeddedServer with a port and a configure block 

### PR DESCRIPTION
**Pull request by vnikolova**
## Summary

Document how to specify port and host when using `embeddedServer` with a `configure` block.

## Details

1. Extended the `embedded-server` snippets project to include an example of `embeddedServer` with a `configure` block and a connector.
2. Configuration in code [[Preview](https://ktor.staging.kotlin-aws.intellij.net/docs/server-configuration-code.html#embedded-engine)]:
- Updated the topic to reference the new code example.
- Added subsections from the "Choose an engine" topic for engine-specific configuration.
3. Configuration in a file [[Preview](https://ktor.staging.kotlin-aws.intellij.net/docs/server-configuration-file.html#config-engine)]:
- Added a new subsection called "Engine configuration" containing relevant content from "Choosing an engine".
4. Choosing an engine [[Preview](https://ktor.staging.kotlin-aws.intellij.net/docs/server-engines.html#configure-engine)]:
- Reuses content from "Configuration in a file" and "Configuration in code" topic.
5. Fixed bad formatting causing warnings.

## Relevant ticket:

[KTOR-7415](https://youtrack.jetbrains.com/issue/KTOR-7415) [Web feedback] Document how to use embeddedServer with a port and a configure block

### CODE REVIEW

File: codeSnippets/snippets/embedded-server/README.md [hunk 1]
```diff
@@ -5,7 +5,14 @@ A sample Ktor project showing how to create a server using the [embeddedServer](
 
 ## Running
 
-To run a sample, execute the following command in a repository's root directory:
+To run the basic sample,execute the following command from the `codeSnippets` directory:
```
Comment (Stexxe) [inline codeSnippets/snippets/embedded-server/README.md:8]: No space after ,

File: codeSnippets/snippets/embedded-server/build.gradle.kts [hunk 2]
```diff
@@ -12,6 +12,14 @@ application {
     mainClass.set("com.example.ApplicationKt")
 }
 
+tasks.register<JavaExec>("runConfiguredServer") {
```
Comment (Stexxe) [inline codeSnippets/snippets/embedded-server/build.gradle.kts:15]: Why is this task necessary?

**Review by Stexxe — APPROVED** [review]

File: codeSnippets/snippets/embedded-server/src/main/kotlin/com/example/Application.kt [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
 import io.ktor.server.engine.*
 import io.ktor.server.netty.*
 
-fun main() {
+/*
+   Important: This contents of this file are referenced by line number in `server-configuration-code.topic`
+    and `server-engines.md`. If you add, remove, or modify any lines, ensure you update the corresponding
+    line numbers in the `code-block` element of the referenced file.
+*/
+
```
Comment (e5l) [inline codeSnippets/snippets/embedded-server/src/main/kotlin/com/example/Application.kt:13]: obsolete empty line

**Review by e5l — APPROVED** [review]

File: codeSnippets/snippets/embedded-server/build.gradle.kts (hunk 2 above)
Comment (vnikolova) [reply codeSnippets/snippets/embedded-server/build.gradle.kts:15]: to run the configured server

**Pull request merged**